### ISSUE TITLE
Why is max_grad_norm set to 0.001 in GRPO defaults?

### ISSUE THREAD

**Issue by pritmish** [PrimeIntellect-ai/verifiers#82]
I'm working on my first GRPO training run and noticed the default gradient clipping value seems unusually low:

```python
def grpo_defaults(run_name: str) -> GRPOConfig:
    return GRPOConfig(
        output_dir=f"outputs/{run_name}",
        run_name=run_name,
        learning_rate=1e-6,
        lr_scheduler_type="constant_with_warmup",
        warmup_steps=10,
        num_train_epochs=1,
        max_steps=500,
        bf16=True,
        max_grad_norm=0.001,  # <-- This seems very low
        ...
    )
```

## Questions:

1. **Why is `max_grad_norm` set to 0.001?** This appears extremely low compared to typical values (0.5-1.0) used in other training scenarios.

2. **Is this intentional for GRPO?** Does GRPO require such aggressive gradient clipping?

3. **Safety of higher values**: Would increasing this to more common values like 0.1 or 1.0 potentially destabilize training, or is 0.001 just a conservative default?

## Context:
- First time doing GRPO
- Looking to understand the rationale behind these hyperparameters
- Want to avoid training instability while ensuring gradients aren't overly constrained

Any insights into the reasoning behind this default or recommendations for tuning would be greatly appreciated!

**Comment by willccbb** [comment]
yes it's intentional, long story, don't have a super intuitive explanation, but generally is more stable in my experience

some discussion/tips in the README, TLDR RL is mysterious and sensitive to HPs and 1.0 is more of a "ML default" than something relevant to GRPO

https://x.com/willccbb/status/1923886079003984085

https://x.com/kalomaze/status/1881957498489581857
